### DISCUSSION TITLE
How to use ui.state with ui.number?

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#1966] [Q&A] [answered]
I'm practicing the official example for ui.state() and make some change to it , It seems the ui.number can't be changed, may be some problem with my code ,or the ui.number its self .  

```py
from nicegui import ui

@ui.refreshable
def counter(name: str):
    with ui.card():
        stat, set_stat = ui.state(0)
        l=ui.label('hhhhh')
        
        ui.number('state',on_change=lambda e: [
            set_stat(e.value),
            l.classes('bg-green') if stat else l.classes('bg-red'),
            print(e.value),
            print('stat:,',stat)
        ])

with ui.row():
    counter('A')
    ui.number()

ui.run()
```

**Comment by falkoschindler** [comment, marked as answer]
That's not how `ui.refreshable` with `ui.state` works. You shouldn't update UI within the event handler, but only use `set_stat` to change the state. Then the `counter` will automatically be re-rendered. All state-dependent styling (e.g. using `.classes`) should be done based on `stat`:

```py
@ui.refreshable
def counter():
    stat, set_stat = ui.state(0)
    ui.label('hhhhh').classes('bg-green' if stat else 'bg-red')
    ui.number('state', value=stat, on_change=lambda e: set_stat(e.value))

counter()
```

**Reply by v479038280** [reply]
We could use if statement in classes🤔

**Reply by falkoschindler** [reply]
The `if` statement is normal Python, nothing special about `.classes`:
https://stackoverflow.com/questions/11880430/how-to-write-inline-if-statement-for-print
